### PR TITLE
Feature/seab 2402/orcid put code

### PR DESCRIPTION
**Pull request by garyluu**
For https://ucsc-cgl.atlassian.net/browse/SEAB-2402 and https://ucsc-cgl.atlassian.net/browse/SEAB-2710

Just modified the existing endpoint to also handle update. Seems easier to use for user too.

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ORCIDHelper.java [hunk 1]
```diff
@@ -121,6 +128,23 @@ public static HttpResponse postWorkString(String baseURL, String id, String work
         }
     }
 
+    /**
+     * This updates an existing ORCID work
+     */
+    public static HttpResponse putWorkString(String baseURL, String id, String workString, String token, String putCode) throws IOException {
+        HttpPut postRequest = new HttpPut(baseURL + id + "/work/" + putCode);
+        postRequest.addHeader("content-type", "application/vnd.orcid+xml");
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ORCIDHelper.java:136]: You could use javax.ws.rs.core.HttpHeaders for header names. Although we so far only use them in a test class.

**Review by coverbeck — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ORCIDHelper.java [hunk 2]
```diff
@@ -121,6 +128,23 @@ public static HttpResponse postWorkString(String baseURL, String id, String work
...
+        postRequest.addHeader("Authorization", "Bearer " + token);
+        StringEntity workEntity = new StringEntity(workString);
+        postRequest.setEntity(workEntity);
+        CloseableHttpClient httpClient = HttpClientBuilder.create().build();
+        try {
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ORCIDHelper.java:141]: A bit cleaner with try catch with resources
https://stackoverflow.com/questions/21574478/what-is-the-difference-between-closeablehttpclient-and-httpclient-in-apache-http

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java [hunk 3]
```diff
@@ -255,19 +258,54 @@ public void exportToORCID(@Parameter(hidden = true, name = "user") @Auth User us
...
+            putCode = entry.getOrcidPutCode();
+        }
         try {
-            String orcidWorkString = ORCIDHelper.getOrcidWorkString(entry, optionalVersion);
-            HttpResponse response = ORCIDHelper.postWorkString(baseApiURL, user.getOrcid(), orcidWorkString,
-                    orcidByUserId.get(0).getToken());
-            if (response.getStatusLine().getStatusCode() != HttpStatus.SC_CREATED) {
-                throw new CustomWebApplicationException("Could not export to ORCID: " + response.getStatusLine().getReasonPhrase(), response.getStatusLine().getStatusCode());
+            String orcidWorkString = ORCIDHelper.getOrcidWorkString(entry, optionalVersion, putCode);
+            HttpResponse response;
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java:268]: Ditto here, think both the client and the response should be closed to be safe. 
Think this implements `CloseableHttpResponse`

**Review by denis-yuen — APPROVED** [review]

**Review by dockstoreNameChange — APPROVED** [review]

**Review by NatalieEO — APPROVED** [review]
Forgot to log back in

**Pull request merged**